### ISSUE TITLE
Missing dependencies- builddatabase and repeatmodeler

### ISSUE THREAD

**Issue by kramergj** [nextgenusfs/funannotate#91]
Hi there,

I am seeming to run into an issue similar to here however,  this fix doesn't seem to work for me.  https://github.com/nextgenusfs/funannotate/issues/13

It seems that everything is setup ok, when I run
`funannotate setup --all`

```
OutputDir    = 
Script mode  = all
Found DB symlinked folder
DB directory set to (/usr/local/share/funannotate), continue [y/n]: y
Creating folder to store DB files: /usr/local/share/funannotate
ln: DB/funannotate: File exists
-----------------------------------------------
Okay, starting downloading of databases....
-----------------------------------------------
MEROPS DB found, skipping download
-----------------------------------------------
UniProt DB found, skipping download. To update delete uniprot_sprot.fasta.
-----------------------------------------------
Pfam-A DB found, skipping download. To update delete Pfam-A.hmm
-----------------------------------------------
PFAM mapping found, skipping download. To update delete Pfam-A.clans.tsv
-----------------------------------------------
dbCAN DB found, skipping download
-----------------------------------------------
Downloading BUSCO outgroups

    The file is already fully retrieved; nothing to do.

-----------------------------------------------
Funannotate repeat protein DB, skipping download
-----------------------------------------------
Gene Ontology already exists, skipping download. To update delete go.obo.
-----------------------------------------------
MIBiG database already exists, skipping download
-----------------------------------------------
InterPro mapping file already exists, skipping download. To update delete interpro.xml
-----------------------------------------------
Checking External Dependencies....
-----------------------------------------------
blastp installed........./usr/local/bin/blastp
hmmsearch installed........./usr/local/bin/hmmsearch
hmmscan installed........./usr/local/bin/hmmscan
augustus installed........./usr/local/bin/augustus
gmes_petap.pl installed........./usr/local/gmes_petap/gmes_petap.pl
mummer installed........./usr/local/bin/mummer
nucmer installed........./usr/local/bin/nucmer
show-coords installed........./usr/local/bin/show-coords
exonerate installed........./usr/local/bin/exonerate
gmap installed........./usr/local/bin/gmap
blat installed........./usr/local/bin/blat
RepeatModeler installed........./usr/local/bin/RepeatModeler
RepeatMasker installed........./usr/local/bin/RepeatMasker
pslCDnaFilter installed........./usr/local/bin/pslCDnaFilter
bedtools installed........./usr/local/bin/bedtools
bamtools installed........./usr/local/bin/bamtools
gag.py installed........./usr/local/bin/gag.py
tbl2asn installed........./usr/local/bin/tbl2asn
braker.pl installed........./usr/local/bin/braker.pl
funannotate installed........./usr/local/bin/funannotate
mafft installed........./usr/local/bin/mafft
trimal installed........./usr/local/bin/trimal
raxmlHPC-PTHREADS installed........./usr/local/bin/raxmlHPC-PTHREADS
tRNAscan-SE installed........./usr/local/bin/tRNAscan-SE
rmOutToGFF3.pl installed........./usr/local/bin/rmOutToGFF3.pl
proteinortho5.pl installed........./usr/local/bin/proteinortho5.pl
-----------------------------------------------
Checking Python modules....
-----------------------------------------------
biopython installed
psutil installed
natsort installed
goatools installed
sklearn installed
matplotlib installed
seaborn installed
numpy installed
pandas installed
-----------------------------------------------
Checking Perl modules....
-----------------------------------------------
Bio::SeqIO installed
Pod::Usage installed
File::Basename installed
threads installed
threads::shared installed
Thread::Queue installed
Getopt::Long installed
FindBin installed
File::Spec installed
File::Path installed
Data::Dumper installed
YAML installed
Carp installed
Hash::Merge installed
Logger::Simple installed
Parallel::ForkManager installed
-----------------------------------------------
Script complete, funannotate is ready to roll!
```

However, when I actually try to run a funannotate test case:
```
funannotate predict -i genome1.fasta -o genome1 -s "Genome one" \
    --isolate fun1 --name GN01_ --augustus_species botrytis_cinerea \
    --protein_evidence proteins.fa --transcript_evidence transcripts.fa --cpus 6

```
This is what I get... 
```
[08:34:25 AM]: Running funannotate v0.6.2
[08:34:26 AM]: Missing Dependencies: BuildDatabase, RepeatModeler.  Please install missing dependencies and re-run script

```
I've tried to run `brew reinstall repeatmodeler` and `brew postinstall repeatmodeler` and all seems to be fine with that, but when I try to run `RepeatModeler -h`  I get...
`-bash: /usr/local/bin/repeatmodeler: /usr/local/bin/perl: bad interpreter: No such file or directory`

Do you have any thoughts on this?

**Comment by nextgenusfs** [comment]
Hi @kramergj, usually the the `bad interpreter` error is referring to the shebang line of the perl script, in this case it is looking for a local version of perl installed at `/usr/local/bin/perl` which is not on your system (likely you are running system Perl, `/usr/bin/perl`).  You should be able to fix the RepeatModeler/RepeatMasker installation by running the manual configuration.  That can be done by moving into the install directory and running `./configure`.  Here's an example:
```
#check your location of perl
$ which perl

#first configure repeatmasker, move into directory, run configure, follow on screen prompts
$ cd /usr/local/Cellar/repeatmasker/4.0.5/libexec/
$ ./configure

#then same thing with repeatmodeler
$ cd /usr/local/Cellar/repeatmodeler/1.0.8/
$ ./configure
```
Note your versions may differ slightly.

**Comment by kramergj** [comment]
Thank you so much for the feedback.  This seems to fix half of the issue, I suppose...

So, yes my perl location is `/usr/bin/perl`. I configured repeatmasker as you described above, and all went well.  

Repeatmodeler is giving me some grief, though.  
I moved into the directory `$ cd /usr/local/Cellar/repeatmodeler/1.0.8_1/` but when I try to configure with `./configure` I am still met with `-bash: ./configure: /usr/local/bin/perl: bad interpreter: No such file or directory`

I don't seem to have anything important missing from the directory
`BuildDatabase				RepeatClassifier
INSTALL_RECEIPT.json			RepeatModeler
Libraries				RepeatUtil.pm
Matrices				SequenceSimilarityMatrix.pm
MultAln.pm				TRFMask
NeedlemanWunschGotohAlgorithm.pm	bin
README					config.txt
Refiner					configure
ReleaseNotes				license.txt
RepModelConfig.pm			util
RepModelConfig.pm.tmpl
`
And although I really don't know if this is useful or relevant, when I open config.txt it seems that the perl path is correct and the content of the file looks like this...
`/usr/bin/perl
/usr/local/Cellar/repeatmodeler/1.0.8_1
/usr/local/opt/repeatmasker/libexec
/usr/local/opt/recon/bin
/usr/local/opt/repeatscout
/usr/local/opt/trf/bin
1
/usr/local/bin
Y
3
`

It seems to me that it is an issue with repeatmodeler, but I have tried reinstalling and it doesn't seem to fix the issue.

**Comment by nextgenusfs** [comment]
I gather that the `configure` program is also a perl script.  You can change the shebang line like this and then re-run configure:
```
sed -i 's,#!/usr/local/bin/perl,#!/usr/bin/perl,g' configure
```
This will change the shebang line in place, you could also just edit in a text editor.  Then try to re-run the ./configure script.

**Comment by kramergj** [comment]
Apparently it is a perl script...

So, from the repeatmodeler directory I did
`sed -i 's,#!/usr/local/bin/perl,#!/usr/bin/perl,g' configure`
It returned `sed: 1: "configure": command c expects \ followed by text`
From that point `./configure` still returns `-bash: ./configure: /usr/local/bin/perl: bad interpreter: No such file or directory`

However, I can get `perl ./configure` to run...
It brings me to a configure prompt similar to that in repeatmasker, where I have to enter the path to several dependency programs.  Set-up works until adding blast, when it fusses about the RepeatMasker library.

`Enter path [  ]: /usr/local/Cellar/blast/2.6.0_1/bin
 
Missing full RepeatMasker repeat library!


  The version of RepeatMasker you have selected was installed
  with the minimal RepeatMasker repeat library.  This library
  is not sufficient for use with RepeatModeler and
  RepeatClassifier.  Please obtain the full repeat database
  from http://www.girinst.org and rerun RepeatMasker/configure
  and the RepeatModeler/configure scripts.`

The repeatmasker configure gives this bit at the end of configure
`Congratulations!  RepeatMasker is now ready to use.
The program is installed with a minimal repeat library
by default.  This library only contains simple, low-complexity,
and common artefact ( contaminate ) sequences.  These are
adequate for use with your own custom repeat library.  If you
plan to search using common species specific repeats you will
need to obtain the complete RepeatMasker repeat library from
GIRI ( www.giriinst.org ) and install it in /usr/local/Cellar/repeatmasker/4.0.5/libexec.
`
Is that what I am supposed to be seeing?  I have installed the RepBase library according to the protocol  given here... https://github.com/nextgenusfs/funannotate/wiki/Mac-OS-Installation

Thanks again for the help!

**Comment by nextgenusfs** [comment]
Okay, a few things then with RepeatModeler versions. There is a new v4.07 from the developers: http://www.repeatmasker.org/RMDownload.html which uses the new RepBase RepeatMasker libraries (version I referenced in instructions above).  I guess the new libraries are not compatible with the older versions of RepeatMasker, i.e. I think home-brew is installing version 4.05 which will require an older version of the RepBase libraries.  So I think either you need to install older Repbase libraries or update to newest RepeatMasker version and use current RepBase library.  Sorry, my instructions were wrong - I didn't know they changed the format of the repeatmasker libraries.  I think the link for the older libraries that will work with v4.05 is here: http://www.girinst.org/server/archive/RepBase21.12/protected/repeatmaskerlibraries/repeatmaskerlibraries-20160829.tar.gz

**Comment by kramergj** [comment]
Yes, that seemed to do the trick! (Currently running, so fingers crossed that is the last of my issues with this!)  Thank you so much for your assistance!

So, just for reference here is a summary...

I downloaded the libraries from the link above and unpacked and copied them to the directory
`/usr/local/Cellar/repeatmasker/4.0.5/libexec/` 
Then I cd into that directory and ran `./configure` and walked through the prompts.  (I had difficulty with the location of RMBlast and so ended up having to cp rmblastn to the same folder as makeblastdb in the blast program).  After configuring repeatmasker, I tried running funannotate again, but was still was met with the dependency issues.  I ended up configuring repeatmodeler (perl ./configure) from the repeatmodeler directory.  After going through the prompts, repeatmodeler was configured with this final message...
`Congratulations!  RepeatModeler is now ready to use.
Simply place:
   /usr/local/Cellar/repeatmodeler/1.0.8_1
in your user's path and review the RepeatModeler documentation here:
   /usr/local/Cellar/repeatmodeler/1.0.8_1/README`
I tried running funannotate again and no dependency errors- seems to be working!

Thank you again!
